test/strings/basic.jl

```diff
@@ -428,6 +428,9 @@ foobaz(ch) = reinterpret(Char, typemax(UInt32))
 # issue #17624, missing getindex method for String
 @test "abc"[:] == "abc"
 
-# PR #18259 reverted because failing test below
-a = [x for x in String([0xcf, 0x83, 0x83, 0x83, 0x83])]
-@test a[1] == 'σ'
+# we assume for performance that next/nextind return values past the end of a
+# string's underlying data; this helps with performance of `done`.
```

Comment:
First, I don't understand why this comment is in the test file at all.  Shouldn't any comments like this be with the `done` method?

Second, I was thinking of something like `# This implementation relies on the fact that next/nextind eventually return values past the end of a String's underlying data`